### PR TITLE
Implement VAPID support

### LINKED ISSUE #120: Add VAPID functionality
Add optional functionality to append [VAPID](https://tools.ietf.org/html/draft-thomson-webpush-vapid-02) to push messages sent by wp-web-push to push servers.

### PR DESCRIPTION
**Pull request by marco-c**
Fixes #120.

### CODE REVIEW

**Comment by marco-c** [comment]
Looks like it's working. I'm using default values for the audience (origin of the WordPress site), subject (email address of the administrator), private key (generated when installing the plugin), so VAPID will be enabled by default.

I'd like some help with the wording (names and description of the fields):
screenshot

@digitarald @groovecoder @jrconlin can you help?

**Comment by groovecoder** [comment]
Wow, that's great to see! Here's the text blurbs I'm using to describe VAPID to dev dashboard users:

> To monitor your push application messages, your application server must implement VAPID. VAPID allows your application server to submit information about itself to the push service, which improves application stability, exception handling, and security.
> 
> See [the VAPID Draft](https://tools.ietf.org/html/draft-thomson-webpush-vapid-02) for more details.

On this push app side of the experience (i.e., generating the VAPID public key), I've added this copy to the [WebPushDataTestPage](https://mozilla-services.github.io/WebPushDataTestPage/):

> To monitor your push messages to Firefox, you need to add your app and its VAPID public key to [the Mozilla push dashboard](https://dev-dashboard.deis.dev.mozaws.net/en/push/apps/). To validate you own the app, you will also need to sign a token with your VAPID private key, and paste the signed token back into the dashboard.
> 
> If you are using this page to test, you can paste this VAPID public key into the dashboard:
> Then paste the app token here to get the signed token value:

This last bit can/should probably be a step-by-step instruction list? I.e. ...

To monitor your push messages to Firefox
1. [Add your app and its VAPID public key to the Mozilla push dashboard](https://dev-dashboard.deis.dev.mozaws.net/en/push/apps/). You will receive a token value to sign
2. Paste the token value here to sign it with your VAPID private key: `<input type="text">`
3. Paste this signed token value back into the dashboard.

**Comment by groovecoder** [comment]
An amazing experience would be a simple button: "Click here to register this push application with the Mozilla Push Dashboard"

File: wp-web-push/wp-web-push-admin.php [hunk 1]
```diff
@@ -349,6 +371,48 @@ public function options() {
 </tr>
 </table>
 
+<?php
+  if (USE_VAPID) {
+?>
+<table class="form-table">
+<h2 class="title"><?php _e('Voluntary Application Server Identification (VAPID)', 'web-push'); ?></h2>
+<p><?php _e('VAPID is useful to monitor your push messages. It allows your server to submit information about itself to the push service, which improves application stability, exception handling, and security.', 'web-push'); ?></p>
+
```
Comment (groovecoder) [inline wp-web-push/wp-web-push-admin.php:378]: Could we always show this top of the VAPID table, regardless of whether or not it's supported by the PHP platform? I.e., include just the description and link to VAPID docs with a message like "VAPID requires PHP 5.5+, mcrypt, gmp, and openssl extensions."

Comment (marco-c) [inline wp-web-push/wp-web-push-admin.php:378]: Done in a1bee7c.

Comment (marco-c) [inline wp-web-push/wp-web-push-admin.php:378]: There's a list now, satisfied prerequisites are green, unsatisfied red.

**Comment by marco-c** [comment]
I'll take care of the integration with a dashboard in a follow-up.

**Pull request merged**